L1Trigger/L1THGCal/src/HGCalTriggerTowerGeometryHelper.cc

```diff
@@ -120,18 +152,98 @@ unsigned short HGCalTriggerTowerGeometryHelper::getTriggerTowerFromEtaPhi(const
+std::unordered_map<unsigned short, float> HGCalTriggerTowerGeometryHelper::getTriggerTower(const l1t::HGCalTriggerSums& thesum) const {
+  std::unordered_map<unsigned short, float> towerIDandShares = {}; 
+  if(!splitModuleSum_){
+    towerIDandShares.insert({getTriggerTowerFromEtaPhi(thesum.position().eta(), thesum.position().phi()), 1.0});
+    return towerIDandShares;
+  } else{
+    HGCalModuleDetId detid(thesum.detId());
+    int moduleU = detid.moduleU();
+    int moduleV = detid.moduleV();
+    int layer = detid.layer();
+    int sector = detid.sector();
+    int zside = detid.zside();
+
+    int subdet = -1;
+    int splitDivisor = -1;
+    if(detid.isHScintillator()){ //FIXME HFNose not handled
+      subdet = 2;
+      layer += 28;
+      splitDivisor = splitDivisorScint_; 
+    } else if(detid.isEE()){
+      subdet = 0;
+      splitDivisor = splitDivisorSilic_;
+    } else if(detid.isHSilicon()) {
+      subdet = 1;
+      layer += 28;
+      splitDivisor = splitDivisorSilic_;
+    }
+    
+    auto module_id_itr = modules_to_trigger_towers_.find(std::to_string(subdet) + " " + 
+                                                         std::to_string(layer) + " " + 
+                                                         std::to_string(moduleU) + " " + 
+                                                         std::to_string(moduleV)
+                                                         );
+    if (module_id_itr != modules_to_trigger_towers_.end()){
+      std::vector<std::string> towerIDandShares_raw; 
+      std::stringstream ss(module_id_itr->second);
+      while(ss.good()){
+      std::string substr;
+      std::getline(ss, substr, ' ');
+      towerIDandShares_raw.push_back(substr);
+      }
+      //eta variables
+      int towerEta = -999;
+      int towerEta_raw = -999;
+      int offsetEta = 2;
+      //phi variables
+      int towerPhi = -999;
+      int towerPhi_raw = -999;    
+      int rotate180Deg = int(nBinsPhi_)/2; 
+      int rotate120Deg = int(nBinsPhi_)/3;
+      int reflectXaxis = int(nBinsPhi_)/2 - 1; //to find phi bin after x -> -x (up to modulo)
+      
+      double towerShare = -999.9; //the share each tower gets from module sum
+     
+      int numTowers = std::stoi(towerIDandShares_raw.at(0));
+      
+      for (int i=0; i<numTowers; i++){
+        towerEta_raw = std::stoi(towerIDandShares_raw.at(3*i+1));
+        towerPhi_raw = std::stoi(towerIDandShares_raw.at(3*i+2));
+        towerShare = std::stod(towerIDandShares_raw.at(3*i+3));
+        
+        towerEta = offsetEta + towerEta_raw; 
+
+        towerPhi = (towerPhi_raw + sector*rotate120Deg + rotate180Deg) % int(nBinsPhi_);//move to the correct sector
+        if(zside==1){
+          towerPhi = reflectXaxis - towerPhi; //correct x -> -x in z>0
+          towerPhi = (int(nBinsPhi_) + towerPhi) % int(nBinsPhi_); // make all phi between 0 to nBinsPhi_-1
+        }
```

Comment:
It might be clearer to implement these lines in a separate function with a name that self-explains what it is doing (applying some transformation on the coordinates).